wire/msgblock.go

```diff
@@ -106,7 +106,19 @@ func (msg *MsgBlock) Deserialize(r io.Reader) error {
 	// At the current time, there is no difference between the wire encoding
 	// at protocol version 0 and the stable long-term storage format.  As
 	// a result, make use of BtcDecode.
-	return msg.BtcDecode(r, 0)
+	//
+	// Passing an encoding type of WitnessEncoding to BtcEncode for
+	// indicates that the transactions within the block are expected to be
+	// serialized according to the new serialization structure defined in
+	// BIP0141.
+	return msg.BtcDecode(r, 0, WitnessEncoding)
```

Comment:
I question whether `Deserialize` should be changed by default.  It should probably take a flag for the encoding instead.  This function, as the comment above indicates, is intended to only be used by the long-term storage format, such as what's in the database.  I'm fairly certain this would cause deserialize existing blocks from disk to fail.

It would also remove the need for a separate `DeserializeNoWitness` function since `BaseEncoding` could be passed.